app/connection/ConnectionManager.js

```diff
@@ -85,7 +85,7 @@ class ConnectionManager extends EventEmitter {
         }
 
         Object.keys(this._pendingRequests).forEach((req_id) => {
-            this._pendingRequests[req_id].reject('Pending requests are rejected as connection is closed.');
+            console.log(`Pending requests are rejected as connection is closed. Request Id= ${req_id}`);
```

Comment:
This leaves the promise unresolved for the lifetime of the promise, which meant that the state is `PENDING`, which is not - we are discarding the promise (which we don't want to do - we want to continue to retry when possible). Better that you catch the error than silently hide it.

Reply:
@brucebinary right, i shouldnt have create a new promise, i send the new req with current promise so it will resolve it when connection back, right?

Reply:
send the existing request, reusing the same promise.

Reply:
@MahboobehMohammadi The log message does not follow what the code is actually doing now.